### PR TITLE
New feature: NUMBEROFGROUPS replacement field on welcome template (#11716)

### PR DESCRIPTION
**Pull request by mfavetti**

### CODE REVIEW

**Comment by Shnoulle** [comment]
See : https://bugs.limesurvey.org/view.php?id=11432

You're right : it can be interesting to have the total groupo number, but : why only in first page ? And wgy not adding STEP too (to show easily something like : you do {STEP} on {NUMBEROFGROUPS} ) or better : TOTALSTEP for exemple.

**Comment by mfavetti** [comment]
Sorry, I didn't know there were pending issues related to this one. I just saw [11716](https://bugs.limesurvey.org/view.php?id=11716) and thought that it would be quick to implement. Since that issue made the reference to the NUMBEROFQUESTIONS replacement, I just did something similar to that. And it was only available on welcome template page. (In the survey helper first page function)

**Comment by Shnoulle** [comment]
Yes : you did right :) . But i'm afraid of the number of core replacement each time an user want a new place.

I think i make a pull request quickly for [11432](https://bugs.limesurvey.org/view.php?id=11432)

Thanks for the mantis link :)

**Comment by LouisGac** [comment]
very good job.
adding replacement keywords is never a problem.
also, avoid doing it with the plugin system for now. It's not correctly implemented (not MVC, not inheriting any Yii class), so they would  make the code more dirty, and harder to maintain.

**Comment by Shnoulle** [comment]
>  adding replacement keywords is never a problem.
I can not be OK with this : all core replacement is done X times, each times it take some ms more, some are reallly used a lot, and some are never used. In fact a lot of our user's don't know the purpose of each replacement : for me some are totally unclear actually.

Adding again and again core replacement make whomle system more complicated ....

**Comment by LouisGac** [comment]
For sure, you're right, we should avoid to add thousands of keyword. 

He just added one keyword. We'll probably add other ones (some have been asked in the forum). That makes custom templating easier for everyone. 

Also, if we speak about performance, be sure that adding more and more dirty core plugin will make more performances problem.

**Comment by Shnoulle** [comment]
PS : and remind all replacement can go to $_SESSION big file : i'm unsure here but must be tested ....

**Comment by jexller** [comment]
Any reason why this PR isn't merged?

**Comment by olleharstedt** [comment]
This PR has a bug. Please always test with debug = 2.

**Comment by LouisGac** [comment]
a notice?

**Comment by mfavetti** [comment]
Sorry, my fault. I do have debug set to 2, but I just didn't happen to see the notice. I made #649 to fix this.

File: application/helpers/replacements_helper.php [hunk 1]
```diff
@@ -683,6 +684,7 @@ function templatereplace($line, $replacements = array(), &$redata = array(), $de
     $coreReplacements['MOVENEXTBUTTON'] = isset($movenextbutton) ? $movenextbutton : '';    // global
     $coreReplacements['NOSURVEYID'] = (isset($surveylist))?$surveylist['nosid']:'';
     $coreReplacements['NUMBEROFQUESTIONS'] = $_totalquestionsAsked;
+    $coreReplacements['NUMBEROFGROUPS'] = $totalgroups;
```
Comment (maziminke) [inline application/helpers/replacements_helper.php]: Where are such placeholders currently documented? I can't find NUMBEROFGROUPS at the manual.

As so often, a feature not documented doesn't exist for the common user.

**Pull request merged**